contracts/BaseStrategy.sol

```diff
@@ -734,118 +421,51 @@ abstract contract BaseStrategy {
             }
             debtPayment = debtOutstanding.sub(loss);
         } else {
-            // Free up returns for Vault to pull
             (profit, loss, debtPayment) = prepareReturn(debtOutstanding);
         }
 
-        // Allow Vault to take up to the "harvested" balance of this contract,
-        // which is the amount it has earned since the last time it reported to
-        // the Vault.
+        uint256 pricePerShare = vault.pricePerShare();
+
         debtOutstanding = vault.report(profit, loss, debtPayment);
 
-        // Check if free returns are left, and re-invest them
+        if (enforeChangeLimit) {
+            uint256 newPricePerShare = vault.pricePerShare();
```

Comment:
i think you need to do a check directly on internals and locked profit because sandwich protection wont give you the real PPS value here